src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -66,72 +64,62 @@ private void initializeTripAndSetVerifiedItinerary(MonitoredTrip monitoredTrip,
                 e
             );
         }
-        ItineraryExistence checkResult = checkItineraryExistence(monitoredTrip, req);
-
-        // Replace the provided trip's itinerary with a verified, non-realtime version of it.
-        if (checkResult != null) {
-            updateTripWithVerifiedItinerary(monitoredTrip, req, checkResult);
-        }
-    }
-
-    /**
-     * Checks that non-realtime itineraries exist for the days the specified monitored trip is active.
-     */
-    private static ItineraryExistence checkItineraryExistence(MonitoredTrip trip, Request request) {
         try {
-            ItineraryExistence checkResult = ItineraryUtils.checkItineraryExistence(trip, false);
-            if (!checkResult.allCheckedDatesAreValid()) {
+            // Check itinerary existence and replace the provided trip's itinerary with a verified, non-realtime
+            // version of it.
+            boolean success = monitoredTrip.checkItineraryExistence(false, true);
+            if (!success) {
                 logMessageAndHalt(
-                    request,
-                    HttpStatus.BAD_REQUEST_400,
-                    "An itinerary does not exist for some of the monitored days for the requested trip."
+                    req,
+                    HttpStatus.INTERNAL_SERVER_ERROR_500,
+                    monitoredTrip.itineraryExistence.message
```

Comment:
Doesn't seem like this should result in a 500 message since nothing faulty happened with the server.